x-pack/test/spaces_api_integration/common/suites/copy_to_space.ts

```diff
@@ -428,7 +436,7 @@ export function copyToSpaceTestSuiteFactory(
     const ambiguousConflictId = `conflict_2_${spaceId}`;
 
     const getResult = (response: TestResponse) => (response.body as CopyResponse).space_2;
-    const expectForbiddenResponse = (response: TestResponse) => {
+    const expectForbiddenESResponse = (response: TestResponse) => {
```

Comment:
Nit: it's not intuitive, but this forbidden response isn't coming from Elasticsearch, it's coming from Kibana. Maybe it would be better to change this function name back to `expectForbiddenResponse` (since this same name is used many other places in the API integration test suites), and change the other function name to something like `expectNoAccessForbiddenResponse`.

Reply:
Ah ok. What's the difference between `noAccess` and not authorized?

Reply:
The Copy-to-space test definitions with both Security and Spaces plugins enabled are here: https://github.com/elastic/kibana/blob/43493bb0d98b948dd254d9698b9fd98fe53e0307/x-pack/test/spaces_api_integration/security_and_spaces/apis/copy_to_space.ts#L169-L200

If you follow the code, the users are defined here: https://github.com/elastic/kibana/blob/43493bb0d98b948dd254d9698b9fd98fe53e0307/x-pack/test/spaces_api_integration/common/lib/authentication.ts#L7

The names are fairly descriptive, but if you continue to follow the code, these users are created with privileges here:

https://github.com/elastic/kibana/blob/43493bb0d98b948dd254d9698b9fd98fe53e0307/x-pack/test/spaces_api_integration/common/lib/create_users_and_roles.ts#L9

So, the two users who have "no access" are those who are able to authenticate with Elasticsearch but do not have any Kibana privileges whatsoever. They'll receive a 403 error at the [API router level](https://github.com/elastic/kibana/blob/a8b1b9e281cc24f97781bc0c3b91f0d09a607060/x-pack/plugins/security/server/authorization/api_authorization.ts#L30-L41).

The other users with either unauthorized read or unauthorized write _do_ have Kibana privileges, so they're able to access this API, but they don't have enough privileges to copy this type of object from Space A to Space B.

Reply:
What I'm puzzled about is why there's a difference in behaviour if you do not have any privileges at all vs. if you have privileges, but just not the right ones. In both cases you don't have access to what you're trying to access, so why isn't the response the same?

I'm sure there's a reason, but I'd prefer if we somehow can name these functions in a way so that reason shows. This way it's easy as a dev to know when one should use which `expect*Response` function. If we go with `expectForbiddenResponse` vs `expectNoAccessForbiddenResponse`, then I'm not sure it's that clear. "No access" and "Forbidden" sounds to me like two ways of saying the same.

Reply:
Talked IRL, and we decided to go with `expectSavedObjectsForbidden` and `expectRouteForbidden` respectively

Reply:
Fixed